pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -852,6 +852,9 @@ func (lb *LbaasV1) EnsureLoadBalancer(clusterName string, apiService *api.Servic
 	for _, host := range hosts {
 		addr, err := getAddressByName(lb.compute, host)
 		if err != nil {
+                        if pool != nil {
```

Comment:
And I think this check is also unnecessary, `pools.Create` should return a none-nil `pool` on success call (Correct me if I'm wrong).